src/zenml/integrations/mlflow/steps/mlflow_deployer.py

```diff
@@ -205,6 +220,132 @@ def mlflow_model_deployer_step(
     return new_service
 
 
+@step(enable_cache=False)
+def mlflow_model_registry_deployer_step(
+    params: MLFlowDeployerParameters,
+) -> MLFlowDeploymentService:
+    """Model deployer pipeline step for MLflow.
+
+    Args:
+        params: parameters for the deployer step
+
+    Returns:
+        MLflow deployment service
+
+    Raises:
+        ValueError: if the registry_model_name is not provided
+        ValueError: if the registry_model_version or registry_model_stage is not provided
+        get_missing_mlflow_experiment_tracker_error: if No MLflow experiment tracker is found
+            in the current active stack
+
+    """
+    if not params.registry_model_name:
+        raise ValueError(
+            "registry_model_name must be provided to the MLflow"
+            "model registry deployer step."
+        )
+    elif not params.registry_model_version and not params.registry_model_stage:
+        raise ValueError(
+            "Either registry_model_version or registry_model_stage must"
+            "be provided in addition to registered_model_name to the MLflow"
+            "model registry deployer step. Since the"
+            "mlflow_model_registry_deployer_step is used in conjunction with"
+            "the mlflow_model_registry."
+        )
+
+    model_deployer = cast(
+        MLFlowModelDeployer, MLFlowModelDeployer.get_active_model_deployer()
+    )
+
+    experiment_tracker = Client().active_stack.experiment_tracker
+    if not isinstance(experiment_tracker, MLFlowExperimentTracker):
+        raise get_missing_mlflow_experiment_tracker_error()
+
+    # fetch the MLflow model registry
+    model_registry = Client().active_stack.model_registry
+    if not isinstance(model_registry, MLFlowModelRegistry):
+        raise ValueError(
+            "The MLflow model registry step can only be used with an "
+            "MLflow model registry."
+        )
+
+    # fetch the model version
+    if params.registry_model_version:
+        model_version = model_registry.get_model_version(
+            name=params.registry_model_name,
+            version=params.registry_model_version,
+        )
+    elif params.registry_model_stage:
+        model_version = model_registry.get_latest_model_versions(
+            name=params.registry_model_name,
+            version_stages=[params.registry_model_stage],
+        )[0]
```

Comment:
This will run into a list index out of range error which isn't nice to read from a user perspective I guess? Maybe an additional check?